pkg/modules/resiliency_test.go

```diff
@@ -57,17 +104,190 @@ func TestResiliencyInjectClusterRole(t *testing.T) {
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
 			success, clusterRole, opConfig, cr := tc(t)
-			newClusterRole, err := ResiliencyInjectClusterRole(clusterRole, cr, opConfig, "node")
+			newClusterRole, err := ResiliencyInjectClusterRole(clusterRole, cr, opConfig, "controller")
+			if success {
+				assert.NoError(t, err)
+				if newClusterRole == nil {
+					panic(err)
+				}
+			} else {
+				assert.Error(t, err)
+			}
+		})
+	}
+}
+func TestResiliencyPrecheck(t *testing.T) {
+	type fakeControllerRuntimeClientWrapper func(clusterConfigData []byte) (ctrlClient.Client, error)
+
+	tests := map[string]func(t *testing.T) (bool, csmv1.Module, csmv1.ContainerStorageModule, ctrlClient.Client, fakeControllerRuntimeClientWrapper){
+		"valid - resiliency module version provided": func(*testing.T) (bool, csmv1.Module, csmv1.ContainerStorageModule, ctrlClient.Client, fakeControllerRuntimeClientWrapper) {
+			customResource, err := getCustomResource("./testdata/cr_powerstore_resiliency.yaml")
+			if err != nil {
+				panic(err)
+			}
+
+			tmpCR := customResource
+			replica := tmpCR.Spec.Modules[0]
```

Comment:
Can we read this config (version) from the same config file i.e. ./testdata/cr_powerstore_resiliency.yaml?
I think this test case will start failing after couple of releases when we will remove the support of v1.6.0.